### PR TITLE
Recursion fix part 2

### PR DESCRIPTION
**Pull request by laurieboyes**
Following on from https://github.com/Financial-Times/next-myft-email/pull/317

Make sure the bit which decides whether to recurse or not knows not to finish recursing too early

### CODE REVIEW

File: functions/get-users/index.js [hunk 1]
```diff
@@ -11,42 +11,48 @@ const metrics = require('./lib-shared/metrics');
...
-	return (users) => {
-		if (!users.length || process.env.NODE_ENV !== 'production') {
-			logger.info('Not putting recursing function whilst in development');
-			return Promise.resolve(users);
-		}
 
-		return recurseFunction({
-			functionName: context.functionName,
-			data: Object.assign({ isRecursion: true }, scheduleEvent)
+	return getUsersResult => Promise.resolve()
```
Comment (debugwand) [inline functions/get-users/index.js]: would this not be better as a fn that does all the things currently inside the first then, and then returns what is currently in the final then call?

File: functions/get-users/index.js [hunk 2]
```diff
@@ -11,42 +11,48 @@ const metrics = require('./lib-shared/metrics');
 const MAX_KINESIS_BATCH_SIZE = 500;
 
 function publishUsers (scheduleEvent) {
-	return (users) => {
-		if (process.env.NODE_ENV === 'development') {
-			logger.info('Not putting onto Kinesis stream while in development env');
-			return Promise.resolve(users);
-		}
+	return getUsersResult => Promise.resolve(getUsersResult.userIds)
```
Comment (debugwand) [inline functions/get-users/index.js]: as other comment, would this not be better as the start of the fn and then you return the promise chain once actually using promises?

File: functions/get-users/index.js [hunk 3]
```diff
@@ -11,42 +11,48 @@ const metrics = require('./lib-shared/metrics');
...
+		.then(users => {
+			if (process.env.NODE_ENV === 'development') {
+				return logger.info('Not putting onto Kinesis stream while in development env');
+			}
 
-		if(!users.length) {
-			return Promise.resolve(users);
-		}
+			if (!users.length) {
+				return logger.info('No users to put onto Kinesis stream in this batch');
```
Comment (debugwand) [inline functions/get-users/index.js]: does the promise not also need to resolve or reject here?

File: functions/get-users/test/index.spec.js [hunk 4]
```diff
@@ -88,13 +91,16 @@ describe('Get users', () => {
...
-	it('calls recurse-function lib if found users', () => {
+	it('calls recurse-function lib if this wasn\'t the final batch', () => {
 		process.env.NODE_ENV = 'production';
 		return invokeLib.then(() => {
 			expect(mocks.recurseFunction.called).to.be.true;
 		});
 	});
 
+	// TODO
+	// it('doesn\'t call recurse-function lib if this was the final batch', () => {
```
Comment (laurieboyes) [inline functions/get-users/test/index.spec.js]: Battled with Sinon for about an hour trying to swap in a different dependency or different internal data for the mocked get-users result. It's a minor test case though so giving up for now

**Pull request merged**